bin/autodetect/Main.cc

```diff
@@ -111,18 +110,19 @@ int main(int argc, char** argv) {
     std::string persistFileName;
     bool isPersistFileNamedPipe{false};
     bool isPersistInForeground{false};
+    bool isPersistDisabled{false};
     std::size_t maxAnomalyRecords{100};
     bool memoryUsage{false};
     bool stopCategorizationOnWarnStatus{false};
     TStrVec clauseTokens;
     if (ml::autodetect::CCmdLineParser::parse(
-            argc, argv, configFile, filtersConfigFile, eventsConfigFile, limitConfigFile,
-            modelConfigFile, logProperties, logPipe, delimiter, lengthEncodedInput,
-            timeField, timeFormat, quantilesStateFile, deleteStateFiles, persistInterval,
+            argc, argv, configFile, filtersConfigFile, eventsConfigFile, modelConfigFile,
+            logProperties, logPipe, delimiter, lengthEncodedInput, timeField,
+            timeFormat, quantilesStateFile, deleteStateFiles, persistInterval,
```

Comment:
Annoyingly the persist interval  is not guaranteed to always be in the job config as it stands at the moment. Further changes to the Java code are required to make this so.

Reply:
Yes, good point.  When the "random time between 3 and 4 hours" default is used it's not in the config.  But actually there's no reason why this default couldn't be moved to the C++ code.  It would simplify things.

The spec is:

1. Write a function that generates a number between 0 and 3599 from the job ID.  One way to do this is to take the first number from a new pseudo-random seeded with a hash of the job ID.
2. Move the default base values for state and quantile persist intervals from Java to C++.
3. If either of these is not set in the job config then create a value by adding the base default to the random adjustment.
4. Add comments to say the idea of the random adjustment based on job ID is that if many jobs are started simultaneously they won't all persist at exactly the same time, thus spreading load on the cluster.